apps/chat/src/containers/TypingIndicator/index.tsx

```diff
@@ -0,0 +1,55 @@
+// Copyright 2020-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+import React, { useEffect } from 'react';
+import styled from 'styled-components';
+import { ChatBubble, } from "amazon-chime-sdk-component-library-react";
+
+import { useChatChannelState, } from "../../providers/ChatMessagesProvider";
+
+import "./index.css";
+
+const TypingIndicatorSection = styled.div`
+  background-color: #f0f1f2;
+  padding: 1rem;
+`;
+
+const TypingIndicator = () => {
+  const {
+    typingIndicator,
+    setTypingIndicator,
+  } = useChatChannelState();
+
+  useEffect(() => {
+    if (typingIndicator) {
+      (async function x() {
+        setTimeout(() => {
+          setTypingIndicator(null);
```

Comment:
Have you tested if a user is typing for more than 30 seconds, does this temporarily cause it to appear as if they're not typing?

Reply:
no, just tested it and the indicator is shown as expected.